merge Spec2 0.7.5 and NewTools 0.4.5

estebanlm opened the pull request:
https://github.com/pharo-spec/NewTools/releases/tag/v0.4.4
https://github.com/pharo-spec/NewTools/releases/tag/v0.4.5
https://github.com/pharo-spec/Spec/releases/tag/v0.7.5

NewTools v0.4.5
- StInspector and StDebugegr now uses a paginator to better show miller columns.
- Some minor bugfixes

NewTools v0.4.4
- new tool SystemReporter (thanks Stef)
- some changes on debugger inspector ui, looking to make things more pleasant.
- fixes https://github.com/pharo-spec/NewTools/issues/120
- fixes https://github.com/pharo-spec/NewTools/issues/113
- fixes https://github.com/pharo-project/pharo/issues/6940
- fixes https://github.com/pharo-project/pharo/issues/7456
- fixes https://github.com/pharo-project/pharo/issues/7453
- fixes https://github.com/pharo-project/pharo/issues/7451

Spec2 v0.7.5
- `SpFilterWithListPresenter` was renamed to `SpFilteringListPresenter`.
- lists and trees now use `displayIcon:` to answer an icon instead `icons:`.
- new presenters `SpPaginatorPresenter` and `SpMillerPaginatorPresenter`.
- Add decoration attributes to `SpStringTableColumn` : `displayColor:`, `displayBackgroundColor:`, `displayBold:`, `displayItalic:`, `displayUnderline:`. API may still change, but this is the easiest to not mess with the Gtk backend.
- several fixes to `SpMillerColumnPresenter`.
- fixes https://github.com/pharo-spec/Spec/issues/952
- fixes https://github.com/pharo-spec/Spec/issues/940 
- fixes https://github.com/pharo-spec/Spec/issues/954 
- fixes https://github.com/pharo-spec/Spec/issues/929 
- fixes https://github.com/pharo-spec/Spec/issues/871 
- ... and several other bugfixes

estebanlm committed:
newtools 0.4.5 intial merge

MarcusDenker commented:
all test runs fail on all platforms with


FileWriteError: File Tests-osx-64-NewTools-Debugger-Tests-Test.xml is closed

estebanlm committed:
fixes

estebanlm committed:
rename usage of icons: (is now displayIcon:)

estebanlm committed:
fixes

estebanlm committed:
add system reporter package

estebanlm committed:
skip suspicious test

estebanlm committed:
skipping the one that seems to be failing

estebanlm committed:
skipping some more tests

estebanlm committed:
skipping more

estebanlm committed:
just remove the class (this is, obviously, temporary)

estebanlm committed:
remove another class

estebanlm committed:
update from trunks (spec and newtools)

estebanlm committed:
missing

estebanlm committed:
removing unnecesary dependences

estebanlm committed:
fixes on URL calling

estebanlm committed:
fix test

estebanlm committed:
removing NewTools-Debugger-Tests for the moment (because some side effect of them seems to be crashing the whole test)

estebanlm committed:
remove spec and ci group

estebanlm committed:
Fix a problem when components (for example playground) was ignoring settings fonts. 
This change allow us to define a #predefinedFont: in the Font property, and font will be taken directly from the one defined in StandardFonts class.

estebanlm committed:
include commander2 before Spec2->Code (because we need it for the commands)

estebanlm committed:
adding missing packages

estebanlm committed:
load comander before spec base

estebanlm committed:
fix tests

estebanlm committed:
do not change this for simulation

estebanlm committed:
restore wrongly removed methods

estebanlm committed:
fix test

estebanlm committed:
remove fuzzy dependencies because of an example. 
add VariablesLibrary to known dependencies (since we need to remove SpTObservable and merge it with TObservable)

estebanlm committed:
Merge branch 'Pharo9.0' into merge-newtools-0.4.5

estebanlm committed:
add VariablesLibrary to known UI dependencies because now Spec2 has started to use them (we will need to remobe SpTObservable and use TObservable)

estebanlm committed:
Merge 0b37ef241447593623a77351bc281a811f9faaef

MarcusDenker reviewed (approved):
Test look good!
